### PR TITLE
Bah z caftercheck

### PR DESCRIPTION
**Pull request by ksantiagoBAH**
## Description
1. On the (Search Results page)[https://staging.va.gov/gi-bill-comparison-tool/profile/11801720], user selects "Other location" in response to "Where will you take the majority of your class", "Please enter the Postal code where you'll take your classes" appears.
2. When user enters an invalid zip (anything other than 5 numbers) and navigates away from the field, the following error message displays: "Zip code must be a 5-digit number"

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/19674

## Testing done
1. Navigate to http://localhost:3001/gi-bill-comparison-tool/profile/11806124
activate voice over
2. Find the question Where will you take the majority of your classes?
3. Select the "An extension campus" radio button
4. Select other from the "Choose the location where you'll take your classes" drop down.
5. Enter less than 5 digits in the Zip code field and tab off
6. Error message "Zip code must be a 5-digit number" will appear

## Screenshots
Screen Shot 2019-09-12 at 9 58 25 AM

## Acceptance criteria
- [x] When user clicks away from the zip code field with an invalid zip code entered, the following error message is displayed: "ZIP Code must be a five digit number"

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
Add unit test coverage for valid/invalid zip code entry validation.

File: src/applications/gi/components/profile/CalculatorForm.jsx [hunk 1]
```diff
@@ -557,11 +566,16 @@ class CalculatorForm extends React.Component {
...
+      const errorMessageCheck =
+        errorMessage !== '' ? errorMessage : inputs.beneficiaryZIPError;
+
       amountInput = (
         <div>
           <ErrorableTextInput
-            errorMessage={inputs.beneficiaryZIPError}
-            label="Please enter the zip code where you’ll take your classes"
+            errorMessage={errorMessageCheck}
+            label="Please enter the Postal code where you'll take your classes"
```
Comment (shawkey-daniel) [inline src/applications/gi/components/profile/CalculatorForm.jsx]: This needs to be zip code not Postal code.

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
The unit test is good. Please also provide an affirmative check for valid zip.

**Review by shawkey-daniel — CHANGES_REQUESTED** [review]
My inline comment still needs resolving. I would change const inputs to const invalidInput and inputs2 to validInput. Remove displayed2, since it is the same as displayed.

File: src/applications/gi/components/profile/CalculatorForm.jsx (hunk 1 above)
Comment (ksantiagomorgan6) [reply src/applications/gi/components/profile/CalculatorForm.jsx]: updated

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by sonntag-adam — APPROVED** [review]

**Review by rianfowler — APPROVED** [review]

**Pull request merged**